ansible_ai_connect/ai/api/model_pipelines/langchain/pipelines.py

```diff
@@ -280,7 +320,86 @@ def __init__(self, config: LANGCHAIN_PIPELINE_CONFIGURATION):
+        outline = params.outline
+        model_id = params.model_id
+        generation_id = params.generation_id
+        file_types = params.file_types
+
+        SYSTEM_MESSAGE_TEMPLATE = """
+        You are an ansible expert optimized to generate Ansible roles.
+        The first print the role name in a way: role_name.
+        Divide section by !-!-!
+        Prefix your comments with the hash character.
+        The second part of the answer is a plain tasks/main.yml file that addresses the user request.
+        Prefix your comments with the hash character.
+        Divide section by !-!-!
+        The third is to prepare a plain defaults/main.yml file based on the first part.
+        This file is a list of variables name and value.
+        Prefix your comments with the hash character.
+        Divide section by !-!-!
+        """
+
+        SYSTEM_MESSAGE_TEMPLATE_WITH_OUTLINE = """
+        You are an ansible expert optimized to generate Ansible roles.
+        The first print the role name in a way: Role name: role_name.
+        Divide section by !-!-!
+        The second part of the answer is a plain tasks/main.yml file that addresses the user request.
+        Prefix your comments with the hash character.
+        Divide section by !-!-!
+        The third is to prepare a plain defaults/main.yml file based on the first part.
+        This file is a list of variables name and value.
+        Prefix your comments with the hash character.
+        Divide section by !-!-!
+        The third part is a step by step explanation of this as a list in the following format:
+        1. first step description
+        2. second step description
+        Never violate the format.
+        Use a new line to explain each step.
+        """
+
+        HUMAN_MESSAGE_TEMPLATE = """
+        This is what the role should do: {text}
+        """
+
+        HUMAN_MESSAGE_TEMPLATE_WITH_OUTLINE = """
+        This is what the role should do: {text}
+        This is a break down of the expected Role: {outline}
+        """
+
+        system_template = (
+            SYSTEM_MESSAGE_TEMPLATE_WITH_OUTLINE if create_outline else SYSTEM_MESSAGE_TEMPLATE
+        )
+        human_template = HUMAN_MESSAGE_TEMPLATE_WITH_OUTLINE if outline else HUMAN_MESSAGE_TEMPLATE
+
+        model_id = self.get_model_id(request.user, None, model_id)
+        llm = self.get_chat_model(model_id)
+
+        chat_template = ChatPromptTemplate.from_messages(
+            [
+                SystemMessagePromptTemplate.from_template(
+                    dedent(system_template),
+                    additional_kwargs={"role": "system"},
+                ),
+                HumanMessagePromptTemplate.from_template(
+                    dedent(human_template), additional_kwargs={"role": "user"}
+                ),
+            ]
+        )
+        chain = chat_template | llm
+        output = chain.invoke({"text": text, "outline": outline})
+        print(">>>>>>>>>>>>>>>>>>>")
```

Comment:
It would be better to log than print (if you want this for any more than just debugging while you developed the change)